Replace resize_dim with set_sizes_and_strides in THTensor_(squeeze)

HuitongQiu opened the pull request:
Summary: Replace resize_dim() with set_sizes_and_strides() in   THTensor_(squeeze) in aten/src/TH/generic/THTensor.cpp

Differential Revision: D14471066

ezyang commented on aten/src/TH/generic/THTensor.cpp:
```diff
@@ -438,20 +438,24 @@ void THTensor_(squeeze)(THTensor *self, THTensor *src)
+        newStride.push_back(src->stride(d));
+      } else {
+        newSize.push_back(self->size(ndim));
+        newStride.push_back(self->stride(ndim));
```
This conditional is very confusing. If `d != ndim`, you use d. But if `d == ndim`, you use ndim. But it's equivalent to use d. You don't need the if statement.

HuitongQiu replied:
I think if statement is still needed. If `d != ndim`, the size/stride of `src` is used, while if `d == ndim`, the size/stride of `self` is used.

ezyang reviewed (changes requested):
I recommend reading the docs to see what `torch.squeeze` does; it will make it clear how you should write this loop.

Commit:
Replace resize_dim with set_sizes_and_strides in THTensor_(squeeze) (#18059)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/18059

Replace resize_dim() with set_sizes_and_strides() in `THTensor_(squeeze)` in aten/src/TH/generic/THTensor.cpp and `THCTensor_(squeeze)` in aten/src/THC/generic/THCTensor.cpp

Differential Revision: D14471066

fbshipit-source-id: 91fc33a9e0e9ed5660bd28011245fe35d1da6800

ezyang reviewed (approved):
Much better!
